src/Dropbox/Authentication/DropboxAuthHelper.php

```diff
@@ -227,6 +227,19 @@ public function getAccessToken($code, $state = null, $redirectUri = null)
         return new AccessToken($accessToken);
     }
 
+    /**
```

Comment:
Instead of updating/mutating the provided access token object, it would be better to simply return a refreshed access token. The method can be refactored to:

```php
    /**
     * Get new Access Token by using the refresh token
     *
     * @param \Kunnu\Dropbox\Models\AccessToken $accessToken - Current access token object
     * @retrun \Kunnu\Dropbox\Models\AccessToken Refreshed Access Token
     */
    public function getRefreshedAccessToken($accessToken)
    {
        $newToken = $this->getOAuth2Client()->getAccessTokenFromRefreshToken($accessToken->getRefreshToken());

        return new AccessToken(
            array_merge(
                $accessToken->getData(),
                $newToken
            )
        );
    }
```

Reply:
### Thank you for reviewing this PR

#### When we make a request for a new `access_token` using the `refresh_token`, we receive following in the response:
  1. `access_token` : It is the new access token generated using the refresh_token
  2. `expires_in` : It is the point of time till which the access_token is valid (In our case, the token remains valid for 4 hours)
  3. `token_type` : Type of token
 
So the two ways to update the expired access_token are :
1. Update these three properties of the current `AcessToken` object (i.e. $accessToken)
2. Copying all the remaining properties of current `AccessToken` object except for `[access_token, expires_in, token_type]` and adding them to the $newToken array.

For step 2, we have to copy in the following manner:
```
if ( $accessToken->uid ) {
            $newToken['uid'] = $accessToken->uid;
        }
```
This is to be done for all existing properties.

And the currently existing object $accessToken is to be deleted as well ( after returning of new AccessToken object).

So why not update the current values rather than returning a whole new object?

Reply:
the following line merges the old/current access token with the new access token which exactly achieves point II: `Copying all the remaining properties of current AccessToken object except for [access_token, expires_in, token_type] and adding them to the $newToken array. Then finally returning a new AcessToken object.`

```php
return new AccessToken(
    array_merge(
        $accessToken->getData(),
        $newToken
    )
);
```

Reply:
Thanks.
Made the desired changes.
Working on the documentation now.